### DISCUSSION TITLE
1.4.0 / 1.0.0-rc9.14

### DISCUSSION THREAD

**Discussion by utpilla** [open-telemetry/opentelemetry-dotnet#4234] [Announcements]
## 1.4.0

This is the stable release encompassing work from 1.4.0-alpha.1 to 1.4.0-rc.4

**OpenTelemetry**

**OpenTelemetry.Api**

**OpenTelemetry.Api.ProviderBuilderExtensions**

**OpenTelemetry.Exporter.Console**

**OpenTelemetry.Exporter.InMemory**

**OpenTelemetry.Exporter.Jaeger**

**OpenTelemetry.Exporter.OpenTelemetryProtocol**

* `AddOtlpExporter` extension methods will now always create a new options instance when named options are NOT used. ([#4200](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4200))

**OpenTelemetry.Exporter.Zipkin**

**OpenTelemetry.Extensions.Hosting**

* Removed deprecated extensions: `AddOpenTelemetryTracing`, `AddOpenTelemetryMetrics`, `Configure`, & `GetServices`. ([#4071](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4071))

**OpenTelemetry.Extensions.Propagators**

## 1.0.0-rc9.14

**OpenTelemetry.Exporter.ZPages**

**OpenTelemetry.Instrumentation.AspNetCore**

**OpenTelemetry.Instrumentation.GrpcNetClient**

**OpenTelemetry.Instrumentation.Http**

**OpenTelemetry.Instrumentation.SqlClient**

**OpenTelemetry.Shims.OpenTracing**

<hr /><em>This discussion was created from the release <a href='https://github.com/open-telemetry/opentelemetry-dotnet/releases/tag/core-1.4.0'>1.4.0 / 1.0.0-rc9.14</a>.</em>

**Comment by ImoutoChan** [comment]
> Removed deprecated extensions: AddOpenTelemetryTracing, AddOpenTelemetryMetrics, Configure, & GetServices. (https://github.com/open-telemetry/opentelemetry-dotnet/pull/4071)

You joined the registration of metrics and traces together, but how to register them separately now?

Your new way is to call:

```csharp
services.AddOpenTelemetry()
  .WithTracing(builder => /* configuration */)
  .WithMetrics(builder => /* configuration */);
```

Can I call them as

```csharp
services.AddOpenTelemetry()
  .WithTracing(builder => /* configuration */);

.... in different part of my code ....

services.AddOpenTelemetry()
  .WithMetrics(builder => /* configuration */);
```

Would it work?

**Reply by utpilla** [reply]
Yes, you could use them in different parts of the code. You could also call them multiple times if you'd like. There will only be one `TracerProvider` or `MeterProvider` created with the configuration provided.

**Comment by wsugarman** [comment]
Is there going to be a `1.4.0` release for [`OpenTelemetry.Exporter.Prometheus.AspNetCore`](https://www.nuget.org/packages/OpenTelemetry.Exporter.Prometheus.AspNetCore)? It's strange that it went from `1.4.0-rc.4` to `1.5.0-alpha.1`.

**Reply by cijothomas** [reply]
_Is there going to be a 1.4.0 release for [OpenTelemetry.Exporter.Prometheus.AspNetCore]_ --__ No

Agree its a bit weird, but Prometheus is treated as a core component (as it is a OTel spec mandated component), and is versioned together with rest of the package. However the Prometheus Exporter spec is experimental, so we cannot release it as 1.4.0 stable (along with the rest).
Hence the awkward jump from 1.4.0-rc.4 to 1.5.0-alpha.1.

**Reply by wsugarman** [reply]
Ah! That makes sense. I didn't realize that the [spec for the exporter itself was experimental](https://opentelemetry.io/docs/reference/specification/metrics/sdk_exporters/prometheus/)!
